folly/String.cpp

```diff
@@ -487,7 +487,7 @@ fbstring errnoStr(int err) {
 #elif FOLLY_HAVE_XSI_STRERROR_R || defined(__APPLE__)
 
   // Using XSI-compatible strerror_r
-  int r = strerror_r(err, buf, sizeof(buf));
+  auto r = strerror_r(err, buf, sizeof(buf));
```

Comment:
This seems like it needs an explanation.

Reply:
not sure how that happened. maybe I fat-fingered a Resharper C++ auto-fix key or something. Fixed.